client/src/components/GrillLightsControls.js

```diff
@@ -9,26 +10,50 @@ import ColorSpeedConfig from 'components/ColorSpeedConfig';
 import useUpdateServer from 'hooks/useUpdateServer';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import ErrorBlock from 'components/ErrorBlock'
 
 export default function GrillLightsControls(props) {
   // constants
-  const DEFAULT_COLORS = {r: props.initialState.red, g: props.initialState.green, b: props.initialState.blue};
   const SAMPLE_DELAY = 250;
+  const STATE_URL = useRef(window.location.origin + '/state?component=animation');
   const ANIMATION_UPDATE_URL = useRef(window.location.origin + '/animation');
+  const GRILLLIGHTSCONTROL_FALLBACK_DEFAULT = useRef({
+    animation: 0,
+    r: 252,
+    g: 101,
+    b: 20,
+    speed: 0.5,
+    direction: 0,
+    density: 0.5,
+    tailLength: 250
+  });
 
   // set up states
-  const [animation, setAnimation] = useState(props.initialState.animation); // enum int: WARM: 0, SOLID: 1, RAINBOW: 2, BREATHE: 3, STROBE: 4, RACER: 5, MARQUEE: 6, MUSIC_MATCH: 7
+  const [animation, setAnimation] = useState(GRILLLIGHTSCONTROL_FALLBACK_DEFAULT.current.animation); // enum int: WARM: 0, SOLID: 1, RAINBOW: 2, BREATHE: 3, STROBE: 4, RACER: 5, MARQUEE: 6, MUSIC_MATCH: 7
   const [color, setColor] = useState({
-    red: DEFAULT_COLORS.r, // int 0-255
-    green: DEFAULT_COLORS.g, // int 0-255
-    blue: DEFAULT_COLORS.b, // int 0-255
+    r: GRILLLIGHTSCONTROL_FALLBACK_DEFAULT.current.r, // int 0-255
+    g: GRILLLIGHTSCONTROL_FALLBACK_DEFAULT.current.g, // int 0-255
+    b: GRILLLIGHTSCONTROL_FALLBACK_DEFAULT.current.b, // int 0-255
   })
-  const [speed, setSpeed] = useState(props.initialState.speed); // float 0.0-1.0
-  const [direction, setDirection] = useState(props.initialState.direction) // int: 0 = left, 1 = right
-  const [density, setDensity] = useState(props.initialState.density) // float 0.0-1.0
-  const [tailLength, setTailLength] = useState(props.initialState.tailLength) // int (0-500)
+  const [speed, setSpeed] = useState(GRILLLIGHTSCONTROL_FALLBACK_DEFAULT.current.speed); // float 0.0-1.0
+  const [direction, setDirection] = useState(GRILLLIGHTSCONTROL_FALLBACK_DEFAULT.current.direction) // int: 0 = left, 1 = right
+  const [density, setDensity] = useState(GRILLLIGHTSCONTROL_FALLBACK_DEFAULT.current.density) // float 0.0-1.0
+  const [tailLength, setTailLength] = useState(GRILLLIGHTSCONTROL_FALLBACK_DEFAULT.current.tailLength) // int (0-500)
+  const [isLoading, setIsLoading] = useState(true);
+  const [errorMsgs, setErrorMsgs] = useState([]);
 
   // set up handlers
+  /**
+   * Adds an array of error strings to the existing state.
+   * @param {Array} errArray Each element is a user-facing error message string.
+   */
+    function updateErrors(errArray) {
+    if (!errArray.some( el => errorMsgs.includes(el))){ // don't duplicate errors. h/t https://stackoverflow.com/questions/16312528/check-if-an-array-contains-any-element-of-another-array-in-javascript
+      setErrorMsgs([...errorMsgs].concat(errArray))
+    }
```

Comment:
Perhaps I'm misunderstanding this, but wouldn't this fail to add a unique error if `errArray` contains one unique error and one error already in `errorMsgs`? Is that desired?

Reply:
yeah I think you are right. here's what I think I need to do instead: check each element in `errArray` to see if it's in `errorMsgs` already, and concatenate it to `errorMsgs` if it's not already there. Right?

Reply:
this is what I did, hard to test so hope it works :)

Reply:
actually tested and... it didn't work. but i reverted back to the existing behavior which did what I hoped... ultimately in practice I'm never calling this function with more than one error in the array anyway. just not going to invest any more time in it for now.